src/main/scala/com/github/jberggg/router/RouterDsl.scala

```diff
@@ -0,0 +1,24 @@
+package com.github.jberggg.router
+
+import fs2.concurrent.SignallingRef
+import org.http4s.Uri.Path
+import Domain._
+
+trait RouterDsl[F[_]] {
+
+    def requestedPaths: SignallingRef[F, Tuple2[Path,HistoryApiState]]
+
+    def navigate(to: Path, withState: HistoryApiState): F[Unit] = requestedPaths.set(Tuple2(to,withState))
+
+}
```

Comment:
I wonder if instead of using `HistoryApiState` as an alias for `js.Any` we can make it a type parameter e.g. `RouterDsl[F[_], S]`. Then it can be typesafe 🤓

Reply:
Just spitballing here, but I wonder whether it could make sense to first have a `History[F, S]` algebra and then implement a `Router[F, S]` on top of that.

Reply:
And then you would try to map the `js.Any` to `S` within the DSL?

Why the two different Algebras, so that you can have a router with / without History API support?

Reply:
> And then you would try to map the `js.Any` to `S` within the DSL?

I think the implementation of the algebra should require an implicit `S <:< js.Any`.

> Why the two different Algebras, so that you can have a router with / without History API support?

Yes, that would definitely be a good reason. It was just a random idea to try and keep the interfaces independent, I need to think more about it :)

Reply:
At the beginning I thought about introducing a trait as a response type of Signal. But then I would have to return a `Either` or `Option` because it's no guaranteed that it is possible to map `js.Any` to the instance/object extending the trait. In the end I decided against that because it would not add any benefit of the user. Leave the handling of the raw `js.Any` by matching to the expected state to the user seemed like the less biased approach to me.

Reply:
Ended up introducing a typed state 😆  However there is a default implementation for receiving the "raw" `js.Any`. Going to describe that in the docs in the future.